### PR TITLE
ARROW-18158: [CI] Use default Python version when installing conda cpp environment to fix conda builds

### PR DESCRIPTION
**Pull request by raulcd**

### CODE REVIEW

**Review by jorisvandenbossche — APPROVED** [review]
I think it is fine to not do this "temporary", as the same will happen with the next new python release

File: ci/docker/conda-cpp.dockerfile [hunk 1]
```diff
@@ -22,13 +22,18 @@ FROM ${repo}:${arch}-conda
...
 COPY ci/conda_env_cpp.txt \
      ci/conda_env_gandiva.txt \
      /arrow/ci/
 RUN mamba install -q -y \
         --file arrow/ci/conda_env_cpp.txt \
         --file arrow/ci/conda_env_gandiva.txt \
+        python=${python} \
         compilers \
         doxygen \
         valgrind && \
```
Comment (kou) [inline ci/docker/conda-cpp.dockerfile:39]: Could you keep this list in alphabetical order?

**Review by kou — APPROVED** [review]
+1

File: ci/docker/conda-cpp.dockerfile (hunk 1 above)
Comment (raulcd) [reply ci/docker/conda-cpp.dockerfile]: Done, thanks @kou

**Comment by jorisvandenbossche** [comment]
The Mac failure is unrelated: https://issues.apache.org/jira/browse/ARROW-18150

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**